notebooks/neural_networks/inception_transferlearning.ipynb

```diff
@@ -0,0 +1,1107 @@
+    "val_summary = ValidationSummary(log_dir='/tmp/bigdl_summaries',\n",
+    "                                        app_name=app_name)\n",
+    "optimizer.set_train_summary(train_summary)\n",
+    "optimizer.set_val_summary(val_summary)\n",
+    "print \"saving logs to \",app_name"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 107,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Optimization Done.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Start to train\n",
+    "trained_model = optimizer.optimize()\n",
+    "print \"Optimization Done.\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 108,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def map_predict_label(l):\n",
+    "    return np.array(l).argmax()\n",
+    "def map_groundtruth_label(l):\n",
+    "    return l[0] - 1\n",
+    "def map_to_label(l):\n",
+    "    return label_nums.keys()[label_nums.values().index(l)]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 109,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "1 ) Ground Truth label:  Fresh potatoes, brown\n",
+      "1 ) Predicted label:  Fresh onions\n",
+      "wrong\n",
+      "2 ) Ground Truth label:  Fresh apples, typical local variety\n",
+      "2 ) Predicted label:  Fresh apple, red delicious\n",
+      "wrong\n",
+      "3 ) Ground Truth label:  Fresh cucumber\n",
+      "3 ) Predicted label:  Fresh oranges\n",
+      "wrong\n",
+      "4 ) Ground Truth label:  Fresh cucumber\n",
+      "4 ) Predicted label:  Fresh cucumber\n",
+      "correct\n",
+      "5 ) Ground Truth label:  Fresh apples, typical local variety\n",
+      "5 ) Predicted label:  Fresh apple, red delicious\n",
+      "wrong\n",
+      "6 ) Ground Truth label:  Fresh cucumber\n",
+      "6 ) Predicted label:  Fresh cucumber\n",
+      "correct\n",
+      "7 ) Ground Truth label:  Fresh carrots\n",
+      "7 ) Predicted label:  Fresh carrots\n",
+      "correct\n",
+      "8 ) Ground Truth label:  Fresh potatoes, brown\n",
+      "8 ) Predicted label:  Fresh potatoes, brown\n",
+      "correct\n"
+     ]
+    }
+   ],
+   "source": [
+    "'''\n",
+    "Look at some predictions and their accuracy\n",
+    "'''\n",
+    "predictions = trained_model.predict(test_rdd)\n",
+    "\n",
+    "num_preds = 8\n",
+    "truth = test_rdd.take(num_preds)\n",
+    "preds = predictions.take(num_preds)\n",
+    "for idx in range(num_preds):\n",
+    "    true_label = str(map_to_label(map_groundtruth_label(truth[idx].label.to_ndarray())))\n",
+    "    pred_label = str(map_to_label(map_predict_label(preds[idx])))\n",
+    "    print idx + 1, ')', 'Ground Truth label: ', true_label\n",
+    "    print idx + 1, ')', 'Predicted label: ', pred_label\n",
+    "    print \"correct\" if true_label == pred_label else \"wrong\""
+   ]
+  },
+  {
```

Comment:
Maybe we should plot the Accuracy-Epoch curve on the validation set to the performance of this classifier